src/networking.c

```diff
@@ -4235,10 +4246,14 @@ int getClientType(client *c) {
 }
 
 static inline int clientTypeIsSlave(client *c) {
-    /* Even though MONITOR clients and ASM destination main channels are marked
+    /* Even though MONITOR clients and ASM destination RDB/main channels are marked
      * as replicas, we want the expose them as normal clients. */
-    if (unlikely((c->flags & CLIENT_SLAVE) && !(c->flags & (CLIENT_MONITOR | CLIENT_REPL_MIGRATION_DEST))))
+    if (unlikely((c->flags & CLIENT_SLAVE) &&
+        !(c->flags & (CLIENT_MONITOR | CLIENT_REPL_MIGRATION_DEST))) &&
+        !(c->slave_req & SLAVE_REQ_SLOTS_SNAPSHOT))
```

Comment:
out of curiosity, why do we need to add rdbchannel here? (not to treat it as a slave).

Reply:
there is a disconnected replica log when rdb channel is closed, i think it is a bit confused, WDYT?

Reply:
No, it is not that confusing. I was just curious. Maybe we can set `CLIENT_REPL_MIGRATION_DEST` on rdbchannel as well, so we won't need to check it with `SLAVE_REQ_SLOTS_SNAPSHOT`